code/mob.dm

```diff
@@ -2968,8 +2968,85 @@
 		var/atom/A = input(usr, "What do you want to pick up?") as() in items
 		A.interact(src)
 
-/mob/proc/can_eat(var/atom/A)
-	return 1
+/// 'A' is the thing being eaten, user is the thing using the thing, and src is of course the thing eating it
+/mob/proc/can_eat(var/obj/item/A, var/mob/user, var/bypass_utensils = 0)
+	if(!istype(A) || !user)
+		return FALSE // who's eating what, now?
+
+	/// Making sure the thing actually exists and isn't cheating the bite-rate. And is also edible, so we don't get people trying to stuff multitools in their mouth
+	if(!src.bioHolder?.HasEffect("mattereater") && GET_COOLDOWN(src, "eat") && (A.edible || A.material?.edible))
+		return src.can_not_eat(A, user, "on_cooldown")
+	else if (!A.amount)
+		return src.can_not_eat(A, user, "all_gone")
+
+	/// Special cases with special circumstances, mostly so awful monsters can eat awful monster food
+	var/edibility_check = SEND_SIGNAL(src, COMSIG_ITEM_CONSUMED_PRE, src, user, A)
+	if(HAS_FLAG(edibility_check, THING_IS_EDIBLE))
+		return TRUE
+
+	/// Now we can check if the thing is actually, in fact, supposed to be edible
+	if(!(A.edible || (A.material && A.material.edible)))
+		return FALSE
+
+	/// Godmode
+	if(src != user && check_target_immunity(src))
+		return src.can_not_eat(A, user, "godmode")
+
+	/// Finally, check if we have proper etiquette
+	if(HAS_FLAG(edibility_check, EATING_NEEDS_A_FORK))
+		return src.can_not_eat(A, user, "lack_fork")
+	if(HAS_FLAG(edibility_check, EATING_NEEDS_A_SPOON))
+		return src.can_not_eat(A, user, "lack_spoon")
+
+	/// And finally finally, if it should actually be edible, do we have somewhere to put it?
+	if (ishuman(src))
+		var/mob/living/carbon/human/H = src
+		var/obj/item/organ/stomach/tummy = H.get_organ("stomach")
+		if (!istype(tummy) || (tummy.broken || tummy.get_damage() > tummy.MAX_DAMAGE))
+			return src.can_not_eat(A, user, "busted_guts")
+
+	/// okay eat the darn thing
+	return TRUE
+
+/// Oh no, we can't eat that! Let's tell the mob why
+/mob/proc/can_not_eat(var/atom/A, var/mob/user, var/fail_reason)
```

Comment:
doesn't this always return false?

Reply:
Yes, it always returns false. I made it a separate proc so it wouldn't clutter the calling proc. Should I move it, or have the calling proc return false instead?

Reply:
just set the default return value at the top

Reply:
I moved the return value out of the proc and made it just send messages to the mob. can_eat returns its own FALSE.